segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -0,0 +1,204 @@
+import numpy as np
+import xarray as xr
+import pandas as pd
+import skimage.measure
+import scipy
+import os
+from scipy.spatial.distance import cdist
+
+
+# Constants for random distance matrix generation
+AB_DIST_MEAN = 100
+AB_DIST_VAR = 1
+
+AC_DIST_MEAN = 20
+AC_DIST_VAR = 1
+
+
+# Constants for random centroid matrix generation
+A_CENTROID_FACTOR = 0.5
+B_CENTROID_FACTOR = 0.9
+C_CENTROID_FACTOR = 0.4
+
+A_CENTROID_COV = [[1, 0], [0, 1]]
+B_CENTROID_COV = [[1, 0], [0, 1]]
+C_CENTROID_COV = [[1, 0], [0, 1]]
+
+
+def direct_init_dist_matrix(num_A=100, num_B=100, num_C=100, distr_AB=None, distr_AC=None, seed=None):
+    """
+    This function will return a random dist matrix such that the distance between cells
+    of types A and B are overall larger than the distance between cells of types A and C
+
+    Each row and column representing a cell.
+    We generate the points using Gaussian distributions
+    Ideally, the parameters for A to B distances will be set such that they produce a lower range of values
+    than A to C distances.
+
+    Will return a random distance matrix as well as the dictionary of associated cell: label IDs
+    The above is generated from the generate_labels function
+
+    Args:
+        num_A: the number of A cells we wish to generate. Default 100
+        num_B: the number of B cells we wish to generate. Default 100
+        num_C: the number of C cells we wish to generate. Default 100
+        distr_AB: if specified, will be a dict listing the mean and variance of the Gaussian distribution
+            we wish to generate numbers from. If None, use the default values.
+        distr_AC: similar to dist_AB. Default will have a higher mean value.
+        seed: whether to fix the random seed or not. Useful for testing.
+            Should be a specified integer value. Default None.
+    """
+
+    # set the mean and variance of the Gaussian distributions of both AB and AC distances
+    if not distr_AB:
+        mean_ab = AB_DIST_MEAN
+        var_ab = AB_DIST_VAR
+    else:
+        mean_ab = distr_AB['mean']
+        var_ab = distr_AB['var']
+
+    if not distr_AC:
+        mean_ac = AC_DIST_MEAN
+        var_ac = AC_DIST_VAR
+    else:
+        mean_ac = distr_AC['mean']
+        var_ac = distr_AC['var']
```

Comment:
referencing the comment on lines 11-25, no need to set new variables; just call `distr_AB` when necessary